src/service-override/extensions.ts

```diff
@@ -445,8 +455,75 @@ export class SimpleExtensionService extends AbstractExtensionService implements
+        this._webExtensionsScannerService.scanUserExtensions(this._userDataProfileService.currentProfile.extensionsResource, { skipInvalidExtensions: true }).then(extensions => user.push(...extensions.map(e => toExtensionDescription(e)))),
+        this._webExtensionsScannerService.scanExtensionsUnderDevelopment().then(extensions => development.push(...extensions.map(e => toExtensionDescription(e, true))))
+      ])
+    } catch (error) {
+      if (error instanceof Error) {
+        this._logService.error(error)
+      }
+    }
+    return dedupExtensions(system, user, development, this._logService)
+  }
+
+  protected async _resolveExtensionsDefault (): Promise<ResolvedExtensions> {
+    const [localExtensions, remoteExtensions] = await Promise.all([
+      this._scanWebExtensions(),
+      this._remoteExtensionsScannerService.scanExtensions()
+    ])
+
+    return new ResolvedExtensions(localExtensions, remoteExtensions, true, true)
+  }
+
   protected override async _resolveExtensions (): Promise<ResolvedExtensions> {
-    return new ResolvedExtensions([], [], false, false)
+    // Taken from vs/workbench/services/extensions/browser/extensionService
+    if (!this._browserEnvironmentService.expectsResolverExtension) {
+      return this._resolveExtensionsDefault()
+    }
+
+    const remoteAuthority = this._environmentService.remoteAuthority!
+
+    // Now that the canonical URI provider has been registered, we need to wait for the trust state to be
+    // calculated. The trust state will be used while resolving the authority, however the resolver can
+    // override the trust state through the resolver result.
+    await this._workspaceTrustManagementService.workspaceResolved
+
+    let resolverResult: ResolverResult
+    try {
+      resolverResult = await this._resolveAuthorityInitial(remoteAuthority)
+    } catch (err) {
+      if (RemoteAuthorityResolverError.isHandled(err)) {
+        console.log('Error handled: Not showing a notification for the error')
+      }
+      this._remoteAuthorityResolverService._setResolvedAuthorityError(remoteAuthority, err)
+
+      // Proceed with the local extension host
+      return this._resolveExtensionsDefault()
+    }
+
+    // set the resolved authority
+    this._remoteAuthorityResolverService._setResolvedAuthority(resolverResult.authority, resolverResult.options)
+    this._remoteExplorerService.setTunnelInformation(resolverResult.tunnelInformation)
+
+    // monitor for breakage
+    const connection = this._remoteAgentService.getConnection()
+    if (connection) {
+      connection.onDidStateChange(async (e) => {
+        if (e.type === PersistentConnectionEventType.ConnectionLost) {
+          this._remoteAuthorityResolverService._clearResolvedAuthority(remoteAuthority)
+        }
+      })
+      connection.onReconnecting(() => this._resolveAuthorityAgain())
+    }
+
+    return this._resolveExtensionsDefault()
```

Comment:
I have the feeling the whole VSCode implementation was copy/pasted now!

Our only need is to be able to provide a custom hostFactory and HostKindPicker. Maybe we can use the VSCode patch to split the SimpleExtensionService in 2 to make it take them as parameters of the constructor. That would allow to better follow VSCode changes

Reply:
> I have the feeling the whole VSCode implementation was copy/pasted now!

Yes indeed, it's practically copy pasted 😄 . It was the best way to align the logic with VSCode when it comes to loading & enabling extensions + handling those edge cases.

So we wouldn't use the `SimpleExtensionService` anymore, but we'd pass in a parameter to the constructor that does the custom `hostFactory` and `HostKindPicker` logic?

Reply:
I'm not very familiar with how the custom extension service works, so I'm wondering if we should do that refactor in a separate PR or this one.

Reply:
I've added 2 commits to your PR, feel free to rewrite it as you wish